[Bug] Failing lint: Need type annotation for "matches"

See `lint` CI job failure on PR #2689.

```
hydra/plugins/completion_plugin.py:110: error: Need type annotation for "matches" (hint: "matches: List[<type>] = ...")  [var-annotated]
```

--- a/hydra/plugins/completion_plugin.py
+++ b/hydra/plugins/completion_plugin.py
@@ -107,7 +107,7 @@ class CompletionPlugin(Plugin):
         if config is None:
             return []
         elif OmegaConf.is_config(config):
-            matches = []
+            matches: List[str] = []
             if word.endswith(".") or word.endswith("="):
                 exact_key = word[0:-1]
                 try:

